### PR TITLE
Enable CI build GitHub Action to run on pull requests on a fork

### PR DESCRIPTION
**Pull request by andyw8**
This wasn't running for external contributors, e.g. https://github.com/Shopify/shopify_app/pull/1187

### CODE REVIEW

**Review by mllemango — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 name: CI
 
-on: 
+on:
   push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (Paulinakhew) [inline .github/workflows/build.yml:7]:
```suggestion
  [push, pull_request]:
    branches: 
      - master
```

Syntax taken from [github docs](https://docs.github.com/en/actions/reference/events-that-trigger-workflows). We can put the duplicates in a list and put `master` by itself

**Review by Paulinakhew — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Paulinakhew) [reply .github/workflows/build.yml:7]: if we don't want to run CI twice, we can do this
```yaml
on:
  push:
    branches-ignore:
      - master
  pull_request:
    branches:
      - master
```

**Comment by andyw8** [comment]
Let's not worry too much about running CI twice for the moment.

**Review by paulomarg — APPROVED** [review]
Agreed, let's run it twice for now and we can change it later.

**Pull request merged**